src/message.rs

```diff
@@ -158,6 +164,9 @@ impl MimeMessage {
             // Content-Type header.
             let mut params = HashMap::new();
             params.insert("boundary".to_string(), self.boundary.clone());
+            if self.message_type == Some(MimeMultipartType::Encrypted) {
+                params.insert("protocol".to_string(), "application/pgp-encrypted".to_string());
```

Comment:
Are all `multipart/encrypted` emails PGP encrypted?

Reply:
No, it is defined in [RFC1847 Section 2.2](https://tools.ietf.org/html/rfc1847#page-6). They can be be for example S/MIME.
But that would require a more complex patch.

Reply:
I'd be more inclined towards a more complex, but flexible, mechanism for allowing message types to have parameters that are controlled from outside a `MimeMessage`.

Reply:
You're right.
Since the `Content-Type` is expected to be set via `message_type`, which is an enum that doesn't allow parameters, should `message_type` be set to `MimeContentTypeHeader`?.
Another way would be to remove `message_type` and expect an outer header with `Content-Type` (or fail).
Both ideas are major changes, do you have a better idea?.

Reply:
I came out with other idea: do not require the `message_type` and use the `Content-Type` header if it's set. This way there're no major changes and the `Content-Type` will be displayed with the original parameters.